scripts/main.js

```diff
@@ -0,0 +1,348 @@
+"use strict";
+document.addEventListener("keydown", (event) => move(event.key));
+
+const rows = document.querySelectorAll("tr");
+
+const button = document.querySelector(".button");
+const score = document.querySelector(".game-score");
+
+let scoreVal = 0;
+
+const board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+const message_start = document.querySelector(".message-start");
+const message_lose = document.querySelector(".message-lose");
+const message_win = document.querySelector(".message-win");
+
+button.addEventListener("click", () => {
+  if (button.classList.contains("start")) {
+    start();
+  } else {
+    restart();
+  }
+});
+
+function start() {
+  button.innerHTML = "Restart";
+  button.classList.add("restart");
+  button.classList.remove("start");
+
+  message_start.classList.add("hidden");
+
+  newCell();
+  newCell();
+  drawBoard();
+}
+
+function restart() {
+  scoreVal = 0;
+  button.innerHTML = "Start";
+  button.classList.add("start");
+  button.classList.remove("restart");
+
+  message_win.classList.add("hidden");
+  message_lose.classList.add("hidden");
+  message_start.classList.remove("hidden");
+
+  for (let y = 0; y < 4; y++) {
+    for (let x = 0; x < 4; x++) {
+      board[y][x] = 0;
+    }
+  }
+  drawBoard();
+}
```

Comment:
The 'restart' function shares common functionality with 'start', such as changing button states and hiding messages. Consider creating shared functions for these operations to reduce code duplication.